src/app/components/add-task/add-task.component.ts

```diff
@@ -19,25 +22,31 @@ export class AddTaskComponent implements OnInit {
     "Task"
   ]
   @ViewChild('addformtemplate') popupContent?: TemplateRef<any>;
-  formGroup: FormGroup | null = null;
+  formGroup!: FormGroup;
+
   constructor(private formBuilder: FormBuilder,
       private dialogRef: MatDialog,
-      private taskService: TaskService) { }
+      private taskService: TaskService,
+      private taskFormValidationService: TaskFormValidationService) { }
 
   ngOnInit(): void {
     this.formGroup = this.formBuilder.group({
-      title: [""],
+      title: ["", {validators: [Validators.required,
+        this.taskFormValidationService.uniqueTitle(),
+      ]}],
       description: [""],
-      type: [""]
+      type: ["", {validators: [Validators.required]}]
     })
   }
 
   submit() {
-    if (this.formGroup) {
```

Comment:
you can pass the form in the submit method directly from the template, will just look cleaner imo